### ISSUE TITLE
Markdoc highlighting and Prettier Markdown formatting at the same time?

### ISSUE THREAD

**Issue by phenomen** [markdoc-extra/vscode-markdoc#7]
When I switch `.mdoc` files to Markdown file association, I lose Markoc highlighting.
When I switch `.mdoc` files to Markdoc file association, I lose Prettier formatting.

Is there a way to have both highlighting and formatting enabled without switching file association all the time?

**Comment by mohitsinghs** [comment]
You can use 
```json
"files.associations": {
  "*.md": "markdoc"
}
```
to enable markdoc highlights in markdown along with prettier formatting. You'll loose preview and other markdown features though.

I just merged #6 and release v0.5.0. This should allow you to use `markdoc.experimental` setting to enable an experimental language server which support formatting provided by markdoc library. 

I am using this for a while but it's experimental and might not work in all scenarios.

**Comment by phenomen** [comment]
> ```json
> "files.associations": {
>   "*.md": "markdoc"
> }
> ```
> 
> to enable markdoc highlights in markdown along with prettier formatting. 

Unfortunately I can't use this as Astro framework requires .mdoc file extension to not mix up Markdoc and Markdown parsers. I will check the new experimental flag, thanks!

**Comment by mohitsinghs** [comment]
I agree with Astro team on that. That's the reason this extension is limited to Markdoc files. Closing this issue. Let me know if it still doesn't work.

Thanks

**Comment by phenomen** [comment]
Highlighting and formatting works! The only issue so far are errors like `Undefined tag: 'aside'(tag-undefined)` for tags that are specified in `markdoc.config.mjs` e.g.:

```js
import { defineMarkdocConfig } from '@astrojs/markdoc/config';
import Aside from '~components/Aside.astro';

export default defineMarkdocConfig({
	tags: {
		aside: {
			render: Aside,
			attributes: {
				variant: { variant: String }
			}
		}
	}
});
```

**Comment by phenomen** [comment]
> The Astro config for Markdoc is a whole different structure altogether which is not supported.

It's just a wrapper for Markdoc config object

- https://markdoc.dev/docs/config
- https://github.com/withastro/astro/blob/main/packages/integrations/markdoc/src/config.ts

Tagging @bholmesdev in case something could be done for Astro compatibility.

**Comment by bholmesdev** [comment]
@phenomen Ah, you're probably getting an error because import aliases are not supported! `markdoc.config` files have the same restrictions as the `astro.config.mjs`. Let me know if using relative paths fixes your problem.

Also, I agree that our Markdoc config mirrors NextJS' pretty much 1-1. Only difference is using `markdoc.config.mjs` instead of `markdoc/config.mjs`. I'm not sure what difficulties you've been hitting there from a language tooling perspective @mohitsinghs. Let me know if we can resolve anything there!

**Comment by mohitsinghs** [comment]
@bholmesdev The primary issue is config structure and loading ESM from CommonJS. At one point I tried shipping a bundler ( esbuild-wasm ) with language server and that worked to some extent but I am still exploring a simpler way.

**Comment by mohitsinghs** [comment]
@phenomen @bholmesdev Released v0.6.0 with bundled esbuild 🎉. This should solve our config related concerns.
